plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidDrawAllocationRefactoring.java

```diff
@@ -0,0 +1,160 @@
+                    ClassInstanceCreation classInstanceCreation = (ClassInstanceCreation) initializer;
+                    InitializerVisitor initializerVisitor = new InitializerVisitor();
+                    initializer.accept(initializerVisitor);
+                    if (initializerVisitor.initializerCanBeExtracted) {
+                        Statement declarationStatement = getAncestor(node, VariableDeclarationStatement.class);
+                        if (declarationStatement != null) {
+                            final ASTBuilder b = this.ctx.getASTBuilder();
+                            final Refactorings r = this.ctx.getRefactorings();
+                            // Deal with collections
+                            if (isMethodBindingSubclassOf(node.getName().resolveTypeBinding(),
+                                    Arrays.asList("AbstractCollection", "Collection", "List", "AbstractMap", "Map"))) {
+                                // It should only works with allocations of
+                                // empty collections.
+                                // Approximation: work only for 0 args
+                                if (classInstanceCreation.arguments().size() == 0) {
+                                    // allocate object outside onDraw
+                                    r.insertBefore(b.move(declarationStatement), onDrawDeclaration);
+                                    // call collection.clear() in the end of onDraw
+                                    ASTNode clearNode = b.toStmt(b.invoke(node.getName().getIdentifier(), "clear"));
+                                    List<Statement> bodyStatements = statements(onDrawDeclaration.getBody());
+                                    Statement lastStatement = bodyStatements.get(bodyStatements.size() - 1);
+                                    if (ASTNode.RETURN_STATEMENT == lastStatement.getNodeType()) {
+                                        r.insertBefore(clearNode, lastStatement);
+                                    } else {
+                                        r.insertAfter(clearNode, lastStatement);
+                                    }
+                                }
+                            } else {
+                                r.insertBefore(b.move(declarationStatement), onDrawDeclaration);
+                            }
+                        }
+                    }
+                }
+            }
+            return VISIT_SUBTREE;
+        }
+    }
+
+    static class InitializerVisitor extends ASTVisitor {
+        private boolean initializerCanBeExtracted = true;
+
+        @Override
+        public boolean visit(MethodInvocation node) {
+            initializerCanBeExtracted = false;
+            return DO_NOT_VISIT_SUBTREE;
+        }
+
+        @Override
+        public boolean visit(SimpleType node) {
+            return DO_NOT_VISIT_SUBTREE;
+        }
```

Comment:
Why this?

I think you should just remove it, or else handle all the possible types (there are plenty ;) )